src/rules/useStrictRule.ts

```diff
@@ -34,61 +35,51 @@ class UseStrictWalker extends Lint.ScopeAwareRuleWalker<{}> {
         return {};
     }
 
-    public visitModuleDeclaration(node: TypeScript.ModuleDeclarationSyntax): void {
-        var childCount = TypeScript.childCount(node.modifiers);
+    public visitModuleDeclaration(node: ts.ModuleDeclaration) {
+        var modifiers = node.modifiers;
+        var hasDeclareModifier = (modifiers != null) && (modifiers.length > 0) && (modifiers[0].kind === ts.SyntaxKind.DeclareKeyword);
 
         // current depth is 2: global scope and the scope created by this module
-        // but skip declare module statements
-        if (this.getCurrentDepth() === 2 &&
-            !(childCount > 0 && TypeScript.childAt(node.modifiers, 0).kind() === TypeScript.SyntaxKind.DeclareKeyword)) {
-
-            if (this.hasOption(UseStrictWalker.OPTION_CHECK_MODULE)) {
-                this.checkUseStrict(node, node.moduleElements);
+        if (this.getCurrentDepth() === 2 && !hasDeclareModifier) {
+            if (node.body != null && this.hasOption(UseStrictWalker.OPTION_CHECK_MODULE)) {
+                this.handleBlock(node, <ts.Block> node.body);
             }
         }
 
         super.visitModuleDeclaration(node);
     }
 
-    public visitFunctionDeclaration(node: TypeScript.FunctionDeclarationSyntax): void {
+    public visitFunctionDeclaration(node: ts.FunctionDeclaration) {
         // current depth is 2: global scope and the scope created by this function
-        if (this.getCurrentDepth() === 2) {
-            if (node.block && this.hasOption(UseStrictWalker.OPTION_CHECK_FUNCTION)) {
-                this.checkUseStrict(node, node.block.statements);
-            }
+        if (this.getCurrentDepth() === 2 &&
+                node.body != null &&
+                this.hasOption(UseStrictWalker.OPTION_CHECK_FUNCTION)) {
+            this.handleBlock(node, node.body);
         }
 
         super.visitFunctionDeclaration(node);
     }
 
-    private checkUseStrict(node: TypeScript.ISyntaxNode, syntaxList: TypeScript.ISyntaxElement[]) {
+    private handleBlock(node: ts.Declaration, block: ts.Block) {
```

Comment:
Oh I see that's why you typed it that way in the cast, above. I think that makes sense.